torchdata/dataloader2/communication/iter.py

```diff
@@ -148,12 +150,43 @@ def DataPipeBehindQueues(source_datapipe, protocol, blocking_request_get=False):
             source_datapipe.reset_iterator()
             protocol.response_reset_iterator()
 
+        elif isinstance(request, communication.messages.PauseRequest):
+            dp_list = list_dps(traverse_dps(source_datapipe))
+            for dp in dp_list:
+                # TODO: Remove this condition after there is `pause` support for round-robin sharding
+                if isinstance(dp, (QueueWrapper, _IterateQueueDataPipes)):
+                    warnings.warn("There is no support for `pause` with round-robin sharding at the moment.")
```

Comment:
When it is dispatching process, the graph has to contain both `QueueWrapper` and `_IterateQueueDataPipes`. Let's check only one of them to trim the warning to once.